mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -333,6 +333,15 @@ def executable_block_runs(self) -> List[BlockRun]:
             )
 
             if dynamic_upstream_block_uuids:
+                if self.allow_blocks_to_fail:
+                    completed_block_uuids = [
```

Comment:
we don't need to calculate it for each block_run. the `completed_block_uuids` above also doesn't need to be calculated for each block_run.

Reply:
It's also better to use `set`